src/Elastic.Markdown/HtmlWriter.cs

```diff
@@ -102,9 +103,27 @@ private async Task<RenderResult> RenderLayout(MarkdownFile markdown, MarkdownDoc
 			fullNavigationRenderResult
 		);
 
-		var currentBaseVersion = legacyPages is { Count: > 0 }
-			? $"{legacyPages.ElementAt(0).Product.VersioningSystem?.Base.Major}.{legacyPages.ElementAt(0).Product.VersioningSystem?.Base.Minor}+"
-			: $"{DocumentationSet.Context.VersionsConfiguration.VersioningSystems[VersioningSystemId.Stack].Base.Major}.{DocumentationSet.Context.VersionsConfiguration.VersioningSystems[VersioningSystemId.Stack].Base.Minor}+";
+		// Acquiring the versioning system for the current page:
+		// 1. If the page has a legacy page mapping, use the versioning system of the legacy page
+		// 2. If the page's docset has a name with a direct product maatch, use the versioning system of the product
+		// 3. If the page's docset has a table of contents entry with a direct product match, use the versioning system of the product
+		// 4. Fallback to the stack versioning system
+		VersioningSystem pageVersioning = null!;
+		if (legacyPages is not null && legacyPages.Count > 0)
+			pageVersioning = legacyPages.ElementAt(0).Product.VersioningSystem!;
+		else if (DocumentationSet.Context.ProductsConfiguration.Products.TryGetValue(DocumentationSet.Name, out var belonging))
+			pageVersioning = belonging.VersioningSystem!;
+		else if (DocumentationSet.Configuration.TableOfContents.FirstOrDefault(t => t is TocReference tRef && !tRef.Source.LocalPath.Equals("/")) is TocReference tocRef)
```

Comment:
This is clever I wonder if its too clever though? What are the cases where this is needed?